scripts/build_dependencies/build_step_tar.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+ const process = require("process");
+ const build_step_axios = require("./build_step_axios");
+ const utils = require('./build_utils');
+
+module.exports = {
+
+    tar : null,
+
+    /**
+     * Will download the file at the given url with the given version to the given path using tar.
+     * Will automatically either use or download the runtime-package for tar as needed.
+     */
+    performStep: async function (url, version, path) {
+        if (utils.npmCheckIfPackageExists("tar") == true) {
+            await this.buildSource(url, version, path);
+        } else {
+            await this.getModuleAndBuildSource(url, version, path);
+        }
+    },
+
+    /**
+     * Will download the file at the given url with the given version to the given path using tar.
+     * Will ALWAYS download tar to the node_modules in scripts/build_dependencies/node_modules.
+     */
+    getPackageAndFetchNativeCode : async function (url, version, path) {
+        const workDir = path.join(__dirname, "../../")
+
+        process.chdir(__dirname);
+        if (this.tar == null) {
+            try {
+                utils.npmDownloadAndInstallRuntimePackage("tar", tar_version);
+                this.tar = require('tar');
+            } catch (error) {
+                console.log("ERROR: Could not download tar! Cannot build CRT");
+                console.log("Please install tar verion " + this.tar_version + " and then run the aws-crt install script again");
+                process.exit(1);
```

Comment:
if this exits, would it leave the user in a different directory?

Reply:
Ah, good catch! Yeah, I need to revert the directory if this happens. 👍